vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,23 +10,158 @@
+        winrt::Microsoft::ReactNative::ReactCoreInjection::GetTopLevelWindowId(m_context.Properties().Handle()));
+
+    m_hwnd = CreateWindow(
+        c_modalWindowClassName,
+        L"React-Native Modal",
+        WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        MODAL_DEFAULT_WIDTH,
+        MODAL_DEFAULT_HEIGHT,
+        roothwnd, // parent
+        nullptr,
+        hInstance,
+        spunk.get());
+  }
+
+  ShowWindow(m_hwnd, SW_NORMAL);
+  BringWindowToTop(m_hwnd);
+  SetFocus(m_hwnd);
+}
+
+void WindowsModalHostComponentView::HideOnUIThread() noexcept {
+  if (m_hwnd) {
+    ::ShowWindow(m_hwnd, SW_HIDE);
+  }
+}
+
+// Windows Procedure - callback function used for handling all messages (generated by NTUser or manual calls to
+// SendMessage)
+LRESULT CALLBACK LogBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+  auto data = reinterpret_cast<::IUnknown *>(GetProp(
+      hwnd,
+      CompHostProperty)); // gets data handle from the property list of specified window (ie the window we want to make)
+  winrt::com_ptr<winrt::IUnknown> spunk;
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result)
+      return result;
+  }
+
+  switch (message) {
+    case WM_NCCREATE: { // sent before WM_CREATE, lparam should be identical to members of CreateWindowEx
+      auto createStruct = reinterpret_cast<CREATESTRUCT *>(lparam); // CreateStruct
+      auto windowData = static_cast<::IUnknown *>(createStruct->lpCreateParams);
+      SetProp(hwnd, CompHostProperty, reinterpret_cast<::IUnknown *>(windowData)); // adds new properties to window
+      break;
+    }
+    case WM_CREATE: { // recieves after window is created but before visible
+      // host.Initialize((uint64_t)hwnd); // this line currently breaks Modal and LogBox
+      break;
+    }
+    case WM_CLOSE: {
+      // Just hide the window instead of destroying it
+      ::ShowWindow(hwnd, SW_HIDE);
+      return 0;
+    }
+    case WM_DESTROY: { // called when we want to destroy the window
+      data->Release();
+      SetProp(hwnd, CompHostProperty, nullptr);
+    }
+  }
+
+  return DefWindowProc(hwnd, message, wparam, lparam);
+}
+
+// Creates and Register a new window class
+void WindowsModalHostComponentView::RegisterWndClass() noexcept {
+  static bool registered = false;
+  if (registered)
+    return;
+
+  HINSTANCE hInstance =
+      GetModuleHandle(NULL); // returns a handle to the file used to create the calling process (.exe file)
+
+  WNDCLASSEXW wcex = {}; // contains window class information
+  wcex.cbSize = sizeof(WNDCLASSEX); // size of windows class (bytes)
```

Comment:
`WNDCLASSEX` instead of `WNDCLASSEXW`?
I think you might actually be able to do `sizeof(wcex)` and it'll do the right thing. I... think.